File: Hot 100/128. 最长连续序列.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        for(auto num: nums){
            st.insert(num);
        }
        int ans=0;
        for(auto num: st){
            if(!st.count(num-1)){
                int len=1;
                int curnum=num;
                while(st.count(curnum+1)){
                    len++;
                    curnum++;
                }
                ans=max(ans,len);
            }
        }
        return ans;
    }
};

File: Hot 100/128. 最长连续序列.go
func longestConsecutive(nums []int) int {
	numSet := map[int]bool{}
	for _, num := range nums {
		numSet[num] = true
	}
	ans := 0
	for num := range numSet {
		if !numSet[num-1] {
			curnum := num
			len := 1
			for numSet[curnum+1] {
				len++
				curnum++
			}
			ans = max(ans, len)
		}
	}
	return ans
}
func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}